SignalMessaging/profiles/OWSProfileManager.m

```diff
@@ -942,7 +947,7 @@ - (void)updateProfileForRecipientId:(NSString *)recipientId
                                          completion:nil];
         }
 
-        if (userProfile.avatarUrlPath.length > 0 && userProfile.avatarFileName.length == 0) {
+        if (avatarUrlPath.length > 0 && !canSkipAvatarUpdate) {
```

Comment:
Isn't this change (and calculating `canSkipAvatarUpdate`) redundant now that we've fixed the avatarUrlPath setter?

The old mental-model, used only the state of the OWSUserProfile model to determine if it needed to download a new avatar.

In other words, anytime we have a non-empty userProfile.avatarFileName, it means we've previously successfully downloaded the avatar specified by userProfile.avatarUrlPath.

Thus, so long as we correctly invalidate avatarFileName upon setting avatarUrlPath, we should be good, right?

Reply:
Good point - we can simplify.  Done, and added a comment.